### PR TITLE
Root LCA Queries Solution

### PR DESCRIPTION
**Pull request by TheGamingMousse**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

### CODE REVIEW

File: solutions/platinum/csa-RootLCAQueries.mdx [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	};
+	dfs(0, -1);
+
+	/** @return d-th ancestor of node u */
+	auto jump = [&](int u, int d) -> int {
+		for (int i = log2dist - 1; i >= 0; i--) {
+			if ((d >> i) & 1) { u = lift[i][u]; }
+		}
+		return u;
+	};
```
Comment (SansPapyrus683) [inline solutions/platinum/csa-RootLCAQueries.mdx:96]: i'll kiss you if you put the lca stuff in a class

Comment (SansPapyrus683) [reply solutions/platinum/csa-RootLCAQueries.mdx:96]: actually uh this pr may go dark for ages bc i was planning on sansifying the bin jumping module

Comment (TheGamingMousse) [reply solutions/platinum/csa-RootLCAQueries.mdx:96]: ill quickly put it in a class

File: solutions/platinum/csa-RootLCAQueries.mdx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+	/** @return d-th ancestor of u */
+	int jump(int u, int d) {
+		for (int i = log2dist - 1; i >= 0; i--) {
+			if ((d >> i) & 1) { u = lift[i][u]; }
+		}
+		return u;
+	}
+
+	/** @return LCA of nodes u and v */
+	int lca(int u, int v) {
```
Comment (SansPapyrus683) [inline solutions/platinum/csa-RootLCAQueries.mdx:91]: i'd argue that these two methods could be made public but do what you want i spose

**Review by dongliu0426 — APPROVED** [review]
nice!

File: solutions/platinum/csa-RootLCAQueries.mdx (hunk 2 above)
Comment (SansPapyrus683) [reply solutions/platinum/csa-RootLCAQueries.mdx:91]: uh if you're going to put this as resolved at least leave a remark :pleading_face:

Comment (TheGamingMousse) [reply solutions/platinum/csa-RootLCAQueries.mdx:91]: oops my bad g

i just put put it as private b/c u aren't gonna need to call it

Comment (TheGamingMousse) [reply solutions/platinum/csa-RootLCAQueries.mdx:91]: altho now that i think about it i might as well just do what u said, will do that rq

**Review by ryanchou-dev — APPROVED** [review]
nice explanation!

**Comment by TheGamingMousse** [comment]
> how did this not get merged alr lol

when i asked on discord u said u wanted to finish sansifying the binlift module

**Pull request merged**